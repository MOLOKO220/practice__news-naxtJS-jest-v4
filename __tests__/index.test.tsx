import { render, screen, act } from "@testing-library/react";
import Home from "@/pages/index";

// Mock fetch
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve([35580845, 35580844]),
  })
) as jest.Mock;

describe("Home", () => {
  // act обертка для компонентов рандарюх повторно
  it("Fetch", async () => {
    await act(() => render(<Home />));

    expect(fetch).toHaveBeenCalledTimes(3);
    // expect(screen.getByText(/35580845/i)).toBeInTheDocument();
  });

  it("Render", async () => {
    render(<Home />);
    expect(screen.getByText(/RESET/i)).toBeInTheDocument();
  });

  it("snapshot", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });

  beforeEach(() => {
    jest.clearAllMocks();
  });
});
