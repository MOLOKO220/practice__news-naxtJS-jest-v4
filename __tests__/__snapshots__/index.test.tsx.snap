// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<div>
  <main
    class="Home"
  >
    <header>
      <button>
        RESET
      </button>
    </header>
    <ul>
      error
    </ul>
  </main>
</div>
`;
